timesketch/ui/graphs/cytoscape.component.spec.ts

```diff
@@ -0,0 +1,264 @@
+}
+
+export const testCytoscapeEvents = [
+  'mousedown', 'mouseup', 'click', 'mouseover', 'mouseout', 'mousemove',
+  'touchstart', 'touchmove', 'touchend', 'tapstart', 'vmousedown',
+  'tapdrag', 'vmousemove', 'tapdragover', 'tapdragout', 'tapend',
+  'vmouseup', 'tap', 'vclick', 'taphold', 'cxttapstart', 'cxttapend',
+  'cxttap', 'cxtdrag', 'cxtdragover', 'cxtdragout', 'boxstart', 'boxend',
+  'boxselect', 'box', 'layoutstart', 'layoutready', 'layoutstop', 'ready',
+  'destroy', 'render', 'resize',
+]
+
+describe('CytoscapeComponent', () => {
+  beforeAll(() => {
+    TestBed.resetTestEnvironment()
+    TestBed.initTestEnvironment(
+      BrowserDynamicTestingModule, platformBrowserDynamicTesting(),
+    )
+  })
+
+  describe('options', () => {
+    let fixture: ComponentFixture<OptionsTestComponent>
+    beforeEach(async () => {
+      TestBed.configureTestingModule({
+        imports: [BrowserModule],
+        declarations: [CytoscapeComponent, OptionsTestComponent],
+      })
+      fixture = TestBed.createComponent(OptionsTestComponent)
+      // prevent 'Error: Timeout - Async callback was not invoked within timeout':
+      fixture.componentInstance.cytoscapeComponent['container'] = undefined
+      await fixture.whenStable()
+    })
+
+    for (const [option, value] of Object.entries(testCytoscapeOptions)) {
+      it(`should handle changes of "${option}" correctly`, () => {
+        fixture.detectChanges()
+        fixture.componentInstance.options[option] = value
+        fixture.detectChanges()
+        const cy = fixture.componentInstance.cytoscapeComponent.cy
+        const newValue = getOptionFromCytoscape(cy, option)
+        expect(newValue).toEqual(value)
+      })
+    }
+
+    for (const option_to_change of Object.keys(testCytoscapeOptions)) {
+      it(`should retain other options after changing "${option_to_change}"`, () => {
+        fixture.componentInstance.options = {...testCytoscapeOptions}
+        fixture.detectChanges()
+        if (option_to_change === 'styleEnabled')
+          fixture.componentInstance.options[option_to_change] = 'true'
+        if (testCytoscapeOptions[option_to_change] === 'additive')
+          fixture.componentInstance.options[option_to_change] = 'single'
+        if (typeof testCytoscapeOptions[option_to_change] === 'boolean')
+          fixture.componentInstance.options[option_to_change] = !fixture.componentInstance.options[option_to_change]
+        if (typeof testCytoscapeOptions[option_to_change] === 'number')
+          fixture.componentInstance.options[option_to_change] = 3.1415
+        fixture.detectChanges()
+        const cy = fixture.componentInstance.cytoscapeComponent.cy
+        for (const [option, value] of Object.entries(testCytoscapeOptions))
+          if (option !== option_to_change) {
+            const actualValue = getOptionFromCytoscape(cy, option)
+            expect(actualValue).toEqual(value)
+          }
+      })
+    }
+  })
+
+  describe('events', () => {
```

Comment:
Why not move this in it's own file, since the concerns are quite separate.

Reply:
Yes, they are separate, but it will clutter the directory and will introduce inconsistency to file names, it will no longer be `x.component.(ts|spec.ts|html)`

Reply:
Ah, you test the same underlying component, then it makes sense to keep them in the same file.